### ISSUE TITLE
Feedback on 6.5.0 Beta

### ISSUE THREAD

**Issue by AdamGrossTX** [maurice-daly/DriverAutomationTool#171]
1. I was using the current release version and the Lenovo XML wouldn't download. I updated to 6.5.0 beta and it fixed the issue.

2. When I launch and select 20H2, I get this error message:
`Loading module from path 'C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\BitsTransfer\BitsTransfer.psd1'.
Loading 'Assembly' from path 'C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\BitsTransfer\Microsoft.BackgroundIntelligentTransfer.Management.Interop.dll'.
Loading 'Assembly' from path 'C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\BitsTransfer\Microsoft.BackgroundIntelligentTransfer.Management.Interop.dll'.
Loading 'FormatsToProcess' from path 'C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\BitsTransfer\BitsTransfer.Format.ps1xml'.
Importing cmdlet 'Add-BitsFile'.
Importing cmdlet 'Remove-BitsTransfer'.
Importing cmdlet 'Complete-BitsTransfer'.
Importing cmdlet 'Get-BitsTransfer'.
Importing cmdlet 'Start-BitsTransfer'.
Importing cmdlet 'Resume-BitsTransfer'.
Importing cmdlet 'Set-BitsTransfer'.
Importing cmdlet 'Suspend-BitsTransfer'.
Exporting cmdlet 'Add-BitsFile'.
Exporting cmdlet 'Remove-BitsTransfer'.
Exporting cmdlet 'Complete-BitsTransfer'.
Exporting cmdlet 'Get-BitsTransfer'.
Exporting cmdlet 'Start-BitsTransfer'.
Exporting cmdlet 'Resume-BitsTransfer'.
Exporting cmdlet 'Set-BitsTransfer'.
Exporting cmdlet 'Suspend-BitsTransfer'.
Importing cmdlet 'Add-BitsFile'.
Importing cmdlet 'Complete-BitsTransfer'.
Importing cmdlet 'Get-BitsTransfer'.
Importing cmdlet 'Remove-BitsTransfer'.
Importing cmdlet 'Resume-BitsTransfer'.
Importing cmdlet 'Set-BitsTransfer'.
Importing cmdlet 'Start-BitsTransfer'.
Importing cmdlet 'Suspend-BitsTransfer'.
Cannot convert value "20H2" to type "System.Int32". Error: "Input string was not in a correct format."
At line:17837 char:5
+ ...               $OSBuild = $WindowsBuildHashTable.Item([int]$OSVersion)
+                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvalidCastFromStringToInteger
 `

**Comment by MichaelGibsonAltrad** [comment]
I can confirm this is the same with us and is caused by the H in the OS version number, yet the variable expects only a number. The code will either have to convert this to 2009 and deal with this conversion in the ApplyDrivers script, or rewrite a lot of code to support the variable type change.

**Comment by MichaelGibsonAltrad** [comment]
Just confirming HP and Lenovo both use the 20H2 label in their filenames so this needs changing to 2009 as the apply driver script does use 2009 to search for and install them. The script does create the package, and we just need to rename the package to include 2009 in the name

**Comment by criiser** [comment]
+1 / BUMP / #attentionseeking - Have the exact same issue. =D

**Comment by MichaelGibsonAltrad** [comment]
Yes, just add 2009 to the package name e.g. "Drivers - HP EliteBook 830 G7 Notebook PC - Windows 10 2009 x64"

I left the folder names the same but what you do is up to you. If you do change the folder name, you'll also need to make sure you update the package data source to point to the renamed folder which will kick off a package update.
